Sources/ComposableArchitecture/Observation/NavigationStack+Observation.swift

```diff
@@ -242,6 +248,60 @@ public struct _NavigationDestinationViewModifier<
   }
 }
 
+@available(iOS 16, macOS 13, tvOS 16, watchOS 9, *)
+public struct MyNavigationLink<Label, P: Identifiable>: View where Label : View {
+  @Environment(\.stackElementGeneration) var stackElementGeneration: Int
```

Comment:
This `View` is required to extract `stackElementGeneration` (the offset in navigation stack).
Naming: TBD